libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -618,6 +627,25 @@ instance HasSpec fn a => Semigroup (Specification fn a) where
 instance HasSpec fn a => Monoid (Specification fn a) where
   mempty = TrueSpec
 
+-- | Try to explain possible causes of (MemberSpec [])
+nullMemberMember :: HasSpec fn a => [a] -> [a] -> Specification fn a
+nullMemberMember xs ys =
+  ErrorSpec
+    ( [ "\nNull intersection of two MemberSpecs"
+      , "  MemberSpec" ++ show xs ++ " <> " ++ "MemberSpec" ++ show ys
+      ]
+        ++ more
+    )
+  where
+    more = case (xs, ys) of
+      ([x], [_]) ->
+        [ "  One possible cause, a term 'x' with two equality predicates, Something like this:"
+        , "  [" ++ show x ++ " ==. x, x ==. y]"
+        , "  (==.) forces the right argument to be solved for first. Causing the conflict"
+        , "  [" ++ show x ++ " ==. x, y ==. x].  Might be solveable depending on how constrained 'y' is."
+        ]
+      _ -> []
+
```

Comment:
This is being addressed in #4317

Reply:
Same as before. Fix it when we rebase on master, after PR #4317 is merged into master.